packages/headless/src/modules/user/repository/user.db.interface.ts

```diff
@@ -0,0 +1,11 @@
+import { Injectable } from '@nestjs/common';
+import { UserEntity } from 'src/entity/user';
+
+@Injectable()
+export abstract class IUserDB {
+  abstract save: (body: UserEntity) => Promise<UserEntity>;
+  abstract findOne: (query: Record<string, string>) => Promise<UserEntity>;
```

Comment:
what is the purpose of Record<string, string>?

Reply:
for find by multiple properties

Reply:
We are writing those codes in typescript. So why are we using this?

Reply:
We are writing those codes in typescript. So why are we using this?